'stddef.h' file not found

Running `hyde` on `test_files/classes.cpp` produces a fatal error:
```
./hyde -hyde-src-root=../test_files -hyde-yaml-dir=test_files_out -hyde-update ../test_files/classes.cpp --
In file included from .../hyde/test_files/classes.cpp:15:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/string:477:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/string_view:176:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__string:56:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/algorithm:639:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/initializer_list:47:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/cstddef:44:15: fatal error: 'stddef.h' file not found
#include_next <stddef.h>
              ^~~~~~~~~~
1 error generated.
Error while processing .../hyde/test_files/classes.cpp.
Error: compilation failed.
```

I'm on macOS Mojave. Changing the hardcoded path in `sources/main.cpp` from `.../clang/9.1.0/include` to `.../clang/10.0.0/include` produces the same error, but about `string.h`:
```
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/string.h:61:15: fatal error: 'string.h' file not found
#include_next <string.h>
              ^~~~~~~~~~
1 error generated.
```

--- a/.gitignore
+++ b/.gitignore
@@ -1,3 +1,4 @@
+.DS_Store
 .vscode/
 build/
 test_files_out/

--- a/generate_test_files.sh
+++ b/generate_test_files.sh
@@ -1,13 +1,20 @@
 #!/bin/bash
 
 EXE_DIR=$(dirname "$0")
-HYDE_DST_ROOT="test_files_out"
 
 pushd ${EXE_DIR} > /dev/null
 
-for CUR_FILE in ${EXE_DIR}/test_files/*; do
+CUR_DUR=$(pwd)
+
+HYDE_PATH=${CUR_DUR}/build/Debug/hyde # This assumes a binary built with Xcode.
+HYDE_SRC_ROOT=${CUR_DUR}
+HYDE_DST_ROOT=${CUR_DUR}/test_files_out
+
+for CUR_FILE in ${CUR_DUR}/test_files/*; do
     echo "Processing $CUR_FILE"
-    ${EXE_DIR}/build/Debug/hyde -hyde-src-root=${EXE_DIR} -hyde-yaml-dir=${HYDE_DST_ROOT} -hyde-update "$CUR_FILE" --
+    CUR_COMMAND="${HYDE_PATH} -hyde-src-root=${HYDE_SRC_ROOT} -hyde-yaml-dir=${HYDE_DST_ROOT} -hyde-update "$CUR_FILE" --"
+    echo $CUR_COMMAND
+    eval $CUR_COMMAND
 done
 
 popd > /dev/null

--- a/sources/main.cpp
+++ b/sources/main.cpp
@@ -161,14 +161,19 @@ int main(int argc, const char** argv) try {
 
     // Get the current Xcode toolchain and add its include directories to the tool.
     std::string xcode_path = exec("xcode-select -p");
-    std::string c_includes = xcode_path + "/usr/lib/llvm-gcc/4.2.1/include";
-    std::string cpp_includes =
-        xcode_path + "/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1";
-    std::string clang_includes = xcode_path + clang_path();
 
-    Tool.appendArgumentsAdjuster(getInsertArgumentAdjuster(("-I" + clang_includes).c_str()));
-    Tool.appendArgumentsAdjuster(getInsertArgumentAdjuster(("-I" + cpp_includes).c_str()));
-    Tool.appendArgumentsAdjuster(getInsertArgumentAdjuster(("-I" + c_includes).c_str()));
+    // Order matters here. The first include path will be looked up first, so should
+    // be the highest priority path.
+    std::string include_directories[] = {
+        xcode_path + clang_path(),
+        "/Library/Developer/CommandLineTools/usr/include/c++/v1",
+        xcode_path + "/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/",
+    };
+
+    for (const auto& include : include_directories) {
+        Tool.appendArgumentsAdjuster(getInsertArgumentAdjuster(("-I" + include).c_str()));
+    }
+
     Tool.appendArgumentsAdjuster(getInsertArgumentAdjuster("-DADOBE_TOOL_HYDE=1"));
 
     if (Tool.run(newFrontendActionFactory(&Finder).get()))

